eventuals/task.h

```diff
@@ -261,29 +314,40 @@ struct _TaskFromToWith final {
     void Dispatch(
         Action action,
         std::optional<MonostateIfVoidOr<From_>>&& from = std::nullopt,
-        std::optional<std::exception_ptr>&& exception = std::nullopt) {
+        std::optional<
+            std::conditional_t<
+                std::tuple_size_v<Catches_>,
+                apply_tuple_types_t<std::variant, Catches_>,
+                std::monostate>>&& error = std::nullopt) {
       CHECK_EQ(value_or_dispatch_.index(), 1u);
 
       std::apply(
-          [&](auto&... args) {
+          [&](auto&&... args) {
             std::get<1>(value_or_dispatch_)(
                 action,
-                std::move(exception),
-                args...,
+                std::move(error),
+                std::forward<decltype(args)>(args)...,
                 std::forward<decltype(from)>(from),
                 e_,
                 *interrupt_,
                 [this](auto&&... args) {
                   k_.Start(std::forward<decltype(args)>(args)...);
                 },
-                [this](std::exception_ptr error) {
-                  k_.Fail(std::move(error));
+                [this](auto&&... void_or_errors) {
+                  if constexpr (sizeof...(void_or_errors)) {
```

Comment:
The previous code always got a `std::exception_ptr`, which means WE SHOULD ALWAYS HAVE HAD AN ERROR!
```suggestion
                [this](auto&& errors) {
```
If there is another place in the code that we need to add a `constexpr` to handle the `void` case then we should, but we shouldn't expect end users that are passing callbacks to us to ever need to do this. Either there should be no errors and they should be able to pass a `void` lambda and know that the code is UNREACHABLE or they should take a `std::variant<Error1, Error2, ...>` and know that there is the possibility of errors.